### PR TITLE
Synchronize clone() behavior with create()

### PR DESCRIPTION
**Pull request by calbach**
- Add explicit free tier billing project status tracking and plumb through to the client
- Fix our representation of the Firecloud billing projects API to match reality
- Move BigQuery role granting from `CreateWorkspace()` -> `initializeUser()`; this seemed a more appropriate place, and I don't think we should be seeing any projects in AoU that we haven't initialized ourselves.
- Write the config file when cloning, as is done during workspace creation

### CODE REVIEW

**Comment by danqrodney** [comment]
The issue with moving this to user initialization is that the bililng project will have just been created and won't have been initialized yet... which means that adding the BQ role right then will fail. (I believe I tried this before.)

See:

https://pmi-engteam.slack.com/archives/C66DX9FD0/p1515618920000520

Since this doesn't have much impact until a user creates a workspace in a project, and calling it multiple times doesn't hurt anything, calling it on create workspace seemed like a reasonable solution.

(Note that we have an outstanding issue that the UI gives you no indication of when the billing project is actually initialized and you can create a workspace. It's not clear whether fixing that would help address this though.)

**Comment by calbach** [comment]
Thanks @danqrodney I figured there was a reason.

How about adding another column to the user table e.g. `billing_project_initialized`, keep trying to grant BigQuery API in `initializeUserIfNeeded()` while unset, then set the field once we succeed? This could also back something in the UI.

**Comment by danqrodney** [comment]
That'd be fine.

**Comment by calbach** [comment]
PTAL, added the proposed column to the user table.

Note: I can't fully test this currently, i.e. creating a new account, due to directory service apparently being busted on my local servers.

**Comment by calbach** [comment]
Yeah, that's the intent. I'm not sure if we will actually need to implement that depending on the details of the researcher sign-up flow, i.e. how long it takes, but seems reasonable to have the information available.

Separately, this is how we've been defining all of our DB enums AFAICT, for better or worse.

**Comment by calbach** [comment]
OK, I've finally been able to validate that this is working locally. PTAL @danqrodney, not sure if this should be reconciled somehow with the "get billingProjects" endpoint, which currently appears to be unused AFAICT.

File: api/src/main/java/org/pmiops/workbench/api/ProfileController.java [hunk 1]
```diff
@@ -218,6 +222,26 @@ private User initializeUserIfNeeded() {
...
+    // On subsequent sign-ins, complete the setup of the FC billing project. There is an
+    // initialization period for Firecloud projects, so it is not possible to do this immediately
+    // after project creation (on first sign-in).
+    if (BillingProjectStatus.PENDING.equals(user.getFreeTierBillingProjectStatus())) {
+      // If the user has not been granted the BQ job user Google role on the billing project yet,
+      // give it to them (so they can run BQ queries from notebooks).
+      try {
+        fireCloudService.grantGoogleRoleToUser(user.getFreeTierBillingProjectName(),
+            FireCloudService.BIGQUERY_JOB_USER_GOOGLE_ROLE, user.getEmail());
+      } catch (ApiException e) {
```
Comment (danqrodney) [inline api/src/main/java/org/pmiops/workbench/api/ProfileController.java]: The assumption here is that as soon as you can grant the role to the user, the billing project is ready for use to create workspaces. That may be true, but I don't know for a fact that this is the case -- you should check with Firecloud folks.

If that isn't the case, we may want to get the billing project status instead here.

Comment (calbach) [reply api/src/main/java/org/pmiops/workbench/api/ProfileController.java]: Good point. I didn't understand that FC was actually intentionally giving us this billing project information as kind of an endpoint to poll. I switched to checking that. I also switched to handle the case where the free tier billing project instead enters the error state.

I've also reconciled this enum with the one we use to report statuses from the billing endpoint. In the future, if we're going to have our own initialization process on workspaces, we may want to maintain some kind of state on our end as to whether or not we have initialization particular billing projects.

Lastly, I fixed a bug in firecloud.yaml; where names did not match those expected.

PTAL

File: api/src/main/java/org/pmiops/workbench/api/ProfileController.java [hunk 2]
```diff
@@ -223,26 +230,57 @@ private User initializeUserIfNeeded() {
...
+        break;
+    }
+
+    // Grant the user BQ job access on the billing project so that they can run BQ queries from
+    // notebooks.
+    try {
+      fireCloudService.grantGoogleRoleToUser(user.getFreeTierBillingProjectName(),
+          FireCloudService.BIGQUERY_JOB_USER_GOOGLE_ROLE, user.getEmail());
+    } catch (ApiException e) {
+      log.log(Level.INFO,
```
Comment (danqrodney) [inline api/src/main/java/org/pmiops/workbench/api/ProfileController.java]: This is no longer expected, right? (You checked status above...)

Seems like we should log severe if this happens, if not return a 500.

Comment (calbach) [reply api/src/main/java/org/pmiops/workbench/api/ProfileController.java]: Yes, forgot to update - updated and made it a warning. I don't want to return a 500 - the approach we've taken so far is that the user should be able to interact with the application, despite their free tier billing project potentially being in a bad state, or still initializing. So a transient error shouldn't block the user anymore than FC's normal delay in project initialization.

**Comment by calbach** [comment]
Added tests, ptal

**Comment by calbach** [comment]
FYI the codacy Java lint was the only failure. We should probably disable/reconfigure that, it's being way too aggressive - I can look into that:

```
Method names should not contain underscores
325
public void testMe_secondCallProjectNotReturned() throws Exception {
```

(in a test, where other test names use this convention).

**Comment by danqrodney** [comment]
LGTM (though you need to rebase).

File: api/db/changelog/db.changelog-master.xml [hunk 3]
```diff
@@ -31,5 +31,7 @@
   <include file="changelog/db.changelog-21.xml"/>
   <include file="changelog/db.changelog-22-email-verification-status.xml"/>
   <include file="changelog/db.changelog-23.xml"/>
+  <include file="changelog/db.changelog-24-email-verification-status-default.xml"/>
```
Comment (calbach) [inline api/db/changelog/db.changelog-master.xml]: Note: Had an interesting merge conflict here, as I had to rebase against change 25 and 24 is checked in but not yet added here. I tested locally that liquibase doesn't seem to care about ordering here (dev-up on master, then dev-up on this branch worked).

**Pull request merged**